src/Splitter.sol

```diff
@@ -0,0 +1,115 @@
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+pragma solidity ^0.8.16;
+
+interface VatLike {
+    function move(address, address, uint256) external;
+    function hope(address) external;
+    function nope(address) external;
+}
+
+interface DaiJoinLike {
+    function vat() external view returns (address);
+    function exit(address, uint256) external;
+}
+
+interface FlapLike {
+    function kick(uint256, uint256) external returns (uint256);
+    function cage(uint256) external;
+}
+
+interface FarmLike {
+    function notifyRewardAmount(uint256 reward) external;
+}
+
+contract Splitter {
+    mapping (address => uint256) public wards;
+    FlapLike    public           flapper;
+    uint256     public           burn; // in WAD; 1 WAD = 100% burn
+
+    VatLike     public immutable vat;
+    DaiJoinLike public immutable daiJoin;
+    FarmLike    public immutable farm;
+
+    event Rely(address indexed usr);
+    event Deny(address indexed usr);
+    event File(bytes32 indexed what, uint256 data);
+    event File(bytes32 indexed what, address data);
+    event Kick(uint256 tot, uint256 lot, uint256 pay);
+
+    constructor(
+        address _daiJoin,
+        address _farm
+    ) {
+        daiJoin = DaiJoinLike(_daiJoin);
+        vat     = VatLike(daiJoin.vat());
+        farm    = FarmLike(_farm);
+        
+        vat.hope(_daiJoin);
+        
+        wards[msg.sender] = 1;
+        emit Rely(msg.sender);
+    }
+
+    modifier auth {
+        require(wards[msg.sender] == 1, "Splitter/not-authorized");
+        _;
+    }
+
+    uint256 internal constant WAD = 10 ** 18;
+    uint256 internal constant RAY = 10 ** 27;
+
+    function rely(address usr) external auth { wards[usr] = 1; emit Rely(usr); }
+    function deny(address usr) external auth { wards[usr] = 0; emit Deny(usr); }
+
+    function file(bytes32 what, uint256 data) external auth {
+        if   (what == "burn") burn = data;
+        else revert("Splitter/file-unrecognized-param");
+        emit File(what, data);
+    }
+
+    function file(bytes32 what, address data) external auth {
+        if (what == "flapper") {
+            vat.nope(address(flapper));
+            flapper = FlapLike(data);
+            vat.hope(data);
+        }
+        else revert("Splitter/file-unrecognized-param");
+        emit File(what, data);
+    }
+
+    // Note: the kick() cadence is determined by flapper.hop()
+    function kick(uint256 tot, uint256) external auth returns (uint256) {
+        vat.move(msg.sender, address(this), tot);
+
+        uint256 lot = tot * burn / WAD;
+        flapper.kick(lot, 0);
```

Comment:
Something we should be aware of is that there could be a case where we might want `burn == 0`.
In that case the expected behavior would be to send everything to the farm, but still enforce the rate limit.
Currently `FlapperUniv2` reverts in this case (getting "revert: UniswapV2: INSUFFICIENT_OUTPUT_AMOUNT").

Probably the minimum we should do is comment that a very low `burn` value will cause the flapper to fail and is not supported. But in general this is a point in favor of moving the rate enforcement to the Splitter (which is probably not something we want to do yet).

Reply:
Added a note to the README and a test checking the splitter around the minimum required burn value for `FlapperUniV2SwapOnly`.

Reply:
Ok, I'm still not 100% at ease with not supporting `burn == 0`, but it might be good enough for this version.
@sunbreak1211 any opinion here?

Reply:
This is a good question. I think we need to implement a rate limiting at the splitter level. And ofc the tighter one will be the one marking the pace.
In the case of burn == 0, we just do not call the Flapper.
Is there a real reason to not do this already? the extra storage read?

Reply:
Yeah to be honest I think the most clean way to do this would be having always the splitter and having the rate limiter there. Then if it should only go to the flapper configured for that. For sure is also a gas waste that configuration.

Reply:
Yes, assuming we change the current implementation I also prefer to have the rate limit only in the Splitter than having it in both places.

Reply:
Ok, for reference [here](https://github.com/makerdao/dss-flappers/pull/12/commits/0e64c8816e0530e9bffa657a233f8383b3a8553a) is an implementation where we have rate-limit in both the Splitter and Flapper but set the flapper.hop() to zero to indicate that the flapper rate limit is effectively not used. 

Next, I'll add a commit with the flapper rate-limiting removed (less wasteful but less generic) and I guess we can decide which one we like best.

Reply:
Ended up opening [another PR](https://github.com/makerdao/dss-flappers/pull/16) for the version without flapper rate-limit as the diff was quite large and I needed to build the changes on top of #14 for the tests to pass.

Reply:
Resolving this comment as we now support `burn == 0`.